File: Exercises/exercise01/ex1.c
//gcc -std=gnu99 -Wall -g -o ex1 ex1.c -lpthread



#include <pthread.h>
#include <stdio.h>



int i = 0;

void* Thread_func1(){
	
	for(int j = 0; j<1000000; j++){
		
		i+=1;
	}
	
	 printf("Value of i - Thread1: %d \n", i);
	
	return NULL;
}

void* Thread_func2(){
	
	for(int j = 0; j<1000000; j++){
		
		i-=1;
	}
	
	 printf("Value of i - Thread2: %d \n", i);

	return NULL;
}



int main(){
	
	pthread_t Thread1;
    pthread_create(&Thread1, NULL, Thread_func1, NULL);
    
    pthread_join(Thread1, NULL);
    
    pthread_t Thread2;
    pthread_create(&Thread2, NULL, Thread_func2, NULL);
    
    pthread_join(Thread2, NULL);
    
	
	return 0;
}

File: Exercises/exercise01/ex1.py
from threading import Thread



i = 0

def Thread_func1():
	
	for j in range(1000000):
		global i
		i+=1
	print(i)
		


print(i)
		
def Thread_func2():
	
	for k in range(1000000):
		global i
		i-=1
	print(i)


def main():
	
	Thread1 = Thread(target = Thread_func1())
	Thread1.start()
    
	Thread2 = Thread(target = Thread_func2())
	Thread2.start()
    
	Thread1.join()
	Thread2.join()
	
	
	

	
main()
